clippy_lints/src/needless_bool.rs

```diff
@@ -206,27 +206,27 @@ impl<'tcx> LateLintPass<'tcx> for BoolComparison {
                     ignore_case,
                     Some((|h| h, "greater than checks against false are unnecessary")),
                     Some((
-                        |h: Sugg<'_>| !h,
+                        |h: Sugg<'tcx>| !h,
                         "less than comparison against true can be replaced by a negation",
                     )),
                     ignore_case,
                     Some((
-                        |l: Sugg<'_>, r: Sugg<'_>| (!l).bit_and(&r),
+                        |l: Sugg<'tcx>, r: Sugg<'tcx>| (!l).bit_and(&r),
                         "order comparisons between booleans can be simplified",
                     )),
                 ),
                 BinOpKind::Gt => check_comparison(
                     cx,
                     e,
                     Some((
-                        |h: Sugg<'_>| !h,
+                        |h: Sugg<'tcx>| !h,
                         "less than comparison against true can be replaced by a negation",
                     )),
                     ignore_case,
                     ignore_case,
                     Some((|h| h, "greater than checks against false are unnecessary")),
                     Some((
-                        |l: Sugg<'_>, r: Sugg<'_>| l.bit_and(&(!r)),
+                        |l: Sugg<'tcx>, r: Sugg<'tcx>| l.bit_and(&(!r)),
```

Comment:
Since the lifetime returned by ```not()``` is no longer ```static```, I have specified a lifetime.
https://github.com/rust-lang/rust-clippy/blob/a7c570df2fba0afc6fee35aa9ff919d4f7192d95/clippy_utils/src/sugg.rs#L464-L466

Reply:
However, my understanding of lifetimes is still limited, so I may be taking a bad approach.

Reply:
Makes sense